src/Sarif/Core/Rule.cs

```diff
@@ -13,5 +13,12 @@ public string Format(string messageId, IEnumerable<string> arguments)
         {
             return string.Format(CultureInfo.CurrentCulture, this.MessageStrings[messageId], arguments.ToArray());
         }
+
+        public bool ShouldSerializeDeprecatedIds()
```

Comment:
>ShouldSerializeDeprecatedIds [](start = 20, length = 28)

the absence of this helper caused a test to fail. somehow the conversion from dictionary to array interacts with json.net default value handling. to be investigated further.

Reply:
Ok

---
In reply to: [249129151](https://github.com/Microsoft/sarif-sdk/pull/1223#discussion_r249129151) [](ancestors = 249129151)